src/main/rx/spektrum.c

```diff
@@ -72,7 +72,7 @@ static uint8_t telemetryBuf[SRXL_FRAME_SIZE_MAX];
 static uint8_t telemetryBufLen = 0;
 
 void srxlRxSendTelemetryDataDispatch(dispatchEntry_t *self);
-static dispatchEntry_t srxlTelemetryDispatch = { .dispatch = srxlRxSendTelemetryDataDispatch};
+static dispatchEntry_t srxlTelemetryDispatch = { .dispatch = srxlRxSendTelemetryDataDispatch, .next = NULL};
```

Comment:
This is not my design, to use the dispatcher with a `dispatchEntry_t`. Not sure how to re-desgn this to use something else. The added initializer is more or less a cosmetic fix. As this PR is mainly a bugfix/improvement of the dispatchAdd() function, I see little or no point in redesigning the Spektrum Telemetry handling. That would be a topic for another PR.

Reply:
Considering that we have discussed fixing this inconsistency about 6 months ago, and there was no action from you on fixing this ever since, I do not think that this is that urgent.

Can I recommend that you start writing that other pull request now, and we can have it supersede this one?

Reply:
Like I pointed out to you the last time we discussed this, you can use the implementation of `src/main/rx/fport.c` as a reference on how to implement this properly.

Reply:
I think this bugfix should go into the 3.5.0 release, to close issue #6348. A redesign using `rxRuntimeConfig.rcProcessFrameFn` would be a perfect candidate for 4.0.

Reply:
A fix of this inconsistency in `spektrum.c` would have been a perfect candidate for 3.3, or even for 3.4, or 3.5. But for some reason you did not seem to think fixing this was urgent back then, so I guess this can wait for 4.0 now.

Reply:
I did not realize a Spektrum Rx redesign was assigned to me, sorry. I really had no time for that anyhow.
But this is really two different issues, in two different areas:
1. Fixing the dispatcher being open for deadlock. ( This PR )
2. Redesigning Spektrum Rx and TM handling.

Reply:
.next is guaranteed to be initialized to zero bytes, explicit initialization is not necessary (and .next is overwritten on enqueue anyway)

Reply:
@mikeller I did not add the dispatching of Spektrum TM. Can we please stop personal accusations of feets being dragged and get a bit more on topic. This PR is a fix for a problem in the dispatcher. A change in the Spektrum code is a totally different issue, please type up a new issue/change request on this. Lets discuss the dispatcher here and the Spektrum code there. 
@ledvinap OK, I can remove that .next initializer. It was a T&E attempt to fix the original issue, that did not help. I just forgot to remove that change when doing the PR.

Reply:
@AndersHoglund: So #5556 is not your pull request then?

I am on the topic of this pull request being a patchup for something that you had been notified of needing fixing and did not find necessary to fix for 6 months. Apologies for calling this 'dragging your feet'.

I think this should have been fixed properly the first time round when it was introduced, and I am not happy to let this slip through now with a patchup, as this makes me suspect that it won't be fixed either before 4.0.

See #6561 as a memory aid.

Reply:
Sure PR 5556 was mine, and now I remember the discussion on re-factoring. And the agreement "Yeah, no problem, this can always be done later." But that was no commitment from my side, a misunderstanding if perceived that way. What I did in that PR was "polishing an old pig." Sure, that pig should be re-factored. But it was not my pig from the beginning. If you want a change, make a request, Plan it, assign it, mile-stone it. I am not a volunteer at this time. 
Great, the request is there in #6561. Lets discuss Spektrum re-factoring there, and dispatcher bug fixing here. Two separate issues.

I do hope you do understand that all this arguing and attacking are draining people from motivation and energy.

Reply:
@AndersHoglund:

> I do hope you do understand that all this arguing and attacking are draining people from motivation and energy.

I know how it feels. I feel the same about people who are pestering others about getting their broken changes merged, on the pretense that they would fix them later on, but then do nothing to fix what they broke for six months, until they are again pestering others to get changes for the same broken code merged again.

So are you going to fix `spektrum.h` before 4.0 now, or are you still going to wait for somebody else to fix your broken stuff?

Reply:
Why arguing on Spektrum issues in a Dispatcher bugfix PR? OT.
OK, I understand how it must have felt being hit by a bugfix PR in the middle of 3.5.0 release work. Unfortunate coincident, sorry about that. No need go get personal because of this. Cool down.
Why do you still think `spektrum.*` is broken by me? I have only added and adjusted things. "Polishing". It was designed to use the dispatcher prior to me getting into this. Not sure I have the energy nor motivation to do anything more on BF. Will need some time to see if I will accept the #6561 at all.

Reply:
@AndersHoglund: Please read #5556. Of course you broke it by introducing inconsistent patterns. Please stop denying this and claiming memory loss.
Also, please stop playing the stalling game.
This pull request should not be merged before we have cleared this up.